torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -645,6 +649,10 @@ def _post_state_dict_hook(
     FSDP module is executed. ``fsdp_state._state_dict_type`` is used to decide
     what postprocessing will be done.
     """
+    if fsdp_state.sharding_strategy == ShardingStrategy.NO_SHARD:
```

Comment:
maybe add a warning saying since no shard is used, full state dict on GPU is returned?